compiler/grainformat/walktree.re

```diff
@@ -121,9 +121,13 @@ let walktree =
 
   let iter_location = (self, location) =>
     if (!List.mem(Code(location), all_locations^)) {
-      all_locations := List.append(all_locations^, [Code(location)]);
+      all_locations := List.cons(Code(location), all_locations^);
     };
 
+  // put it in the right order ready for the final sort as hopefully
+  // almost the right order is faster to sort
+  all_locations := List.rev(all_locations^);
```

Comment:
IIRC OCaml default sort implementation is mergesort, so this wouldn't help.

Reply:
removed it